### PR TITLE
Footsteps without view bobbing

### PR DESCRIPTION
**Pull request by desttinghim**
This PR allows the footstep sounds to play without the view bobbing at the same time, as per bug #5602. It also removes the redundant view_bobbing setting from minetest.conf files.

### CODE REVIEW

**Comment by desttinghim** [comment]
Newest commit (796020b) adds documentation

File: src/camera.cpp [hunk 1]
```diff
@@ -280,7 +279,7 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime,
 	v3f rel_cam_target = v3f(0,0,1);
 	v3f rel_cam_up = v3f(0,1,0);
 
-	if (m_view_bobbing_anim != 0 && m_camera_mode < CAMERA_MODE_THIRD)
+	if (m_cache_view_bobbing_amount != 0 && m_view_bobbing_anim != 0 && m_camera_mode < CAMERA_MODE_THIRD)
```
Comment (paramat) [inline src/camera.cpp]: Because '_amount' is a float maybe it should be '!= 0.0f'? Not sure.

**Comment by paramat** [comment]
Thanks for updating the settings translation file, there's no need to update all the translations files as those are updated only every 1-2 months or so, but thanks anyway.

File: src/camera.cpp (hunk 1 above)
Comment (desttinghim) [inline src/camera.cpp]: Probably a good idea. I'll change it.

**Comment by desttinghim** [comment]
Changed to using 0.0f for comparison instead of an int.

File: src/camera.cpp [hunk 2]
```diff
@@ -280,7 +279,7 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime,
 	v3f rel_cam_target = v3f(0,0,1);
 	v3f rel_cam_up = v3f(0,1,0);
 
-	if (m_view_bobbing_anim != 0 && m_camera_mode < CAMERA_MODE_THIRD)
+	if (m_cache_view_bobbing_amount != 0.0f && m_view_bobbing_anim != 0.0f && m_camera_mode < CAMERA_MODE_THIRD)
```
Comment (paramat) [inline src/camera.cpp]: please split the line at 80 columns (with tab size 4)

**Comment by paramat** [comment]
+1 once long line is split.

**Comment by desttinghim** [comment]
Long line has been split.

**Comment by paramat** [comment]
:+1:

File: src/camera.cpp [hunk 3]
```diff
@@ -280,7 +279,8 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime,
 	v3f rel_cam_target = v3f(0,0,1);
 	v3f rel_cam_up = v3f(0,1,0);
 
-	if (m_view_bobbing_anim != 0 && m_camera_mode < CAMERA_MODE_THIRD)
+	if (m_cache_view_bobbing_amount != 0.0f && m_view_bobbing_anim != 0.0f &&
+		m_camera_mode < CAMERA_MODE_THIRD)
 	{
```
Comment (nerzhul) [inline src/camera.cpp]: brace on wrong line

File: src/camera.cpp [hunk 4]
```diff
@@ -467,7 +467,6 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime,
 	const bool swimming = (movement_XZ || player->swimming_vertical) && player->in_liquid;
 	const bool climbing = movement_Y && player->is_climbing;
 	if ((walking || swimming || climbing) &&
-			m_cache_view_bobbing &&
 			(!g_settings->getBool("free_move") || !m_client->checkLocalPrivilege("fly")))
 	{
```
Comment (nerzhul) [inline src/camera.cpp]: place move this brace too

**Comment by desttinghim** [comment]
Moved braces :+1:

**Pull request merged**